WebDriverAgentTests/UnitTests/XCTestElementSnapshot.h

```diff
@@ -0,0 +1,14 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <WebDriverAgentLib/XCElementSnapshot.h>
+
+@interface XCTestElementSnapshot : XCElementSnapshot
```

Comment:
Definitely not, this name is confusing and you shouldn't really inherit from `XCElementSnapshot`. I would name it `XCElementDouble` that implements `FBElement` protocol and make your method in `FBXPath` accept `id<FBElement>` instead of `XCElementSnapshot`. Then you can create lightweight elements for testing.